### PR TITLE
Update to new memory model

### PR DESCRIPTION
**Pull request by e5l**

### CODE REVIEW

File: gradle.properties [hunk 1]
```diff
@@ -23,10 +23,11 @@ kotlin_version=1.6.0
 kotlin.native.ignoreDisabledTargets=true
 kotlin.mpp.stability.nowarn=true
 kotlin.mpp.enableCompatibilityMetadataVariant=true
+kotlin.native.binary.memoryModel=experimental
```
Comment (hfhbd) [inline gradle.properties:26]: What happens, if the consumer of Ktor does not use the experimental MM? Do you need a check, like a `native-mt` one too?

Comment (e5l) [reply gradle.properties:26]: Nope, we drop native-mt support

Comment (hfhbd) [reply gradle.properties:26]: I know, I mean a check: `isExperimentalMMActive`

Comment (e5l) [reply gradle.properties:26]: I'm not sure that if will be possible, it's likely to fail with `InvalidImmutablityException`

**Comment by e5l** [comment]
```kotlin
#0  0x00007ffff7d2ff1b in sched_yield () at ../sysdeps/unix/syscall-template.S:120
#1  0x0000000000a51972 in kotlin::mm::SuspendThreads() ()
#2  0x0000000000a58c0b in kotlin::gc::SameThreadMarkAndSweep::PerformFullGC() ()
#3  0x0000000000a58953 in kotlin::gc::SameThreadMarkAndSweep::ThreadData::PerformFullGC() ()
#4  0x0000000000a588d9 in kotlin::gc::SameThreadMarkAndSweep::ThreadData::SafePointRegularSlowPath() ()
#5  0x0000000000a5dc0a in kotlin::gc::SameThreadMarkAndSweep::ThreadData::SafePointLoopBody() ()
#6  0x0000000000888e8f in kfun:kotlin.collections#singleOrNull__at__kotlin.collections.Iterable<0:0>(kotlin.Function1<0:0,kotlin.Boolean>){0§<kotlin.Any?>}0:0? ()
    at /mnt/agent/work/6326934d18cfe24e/kotlin/libraries/stdlib/common/src/generated/_Collections.kt:661
#7  <inlined-out:<anonymous>> () at /home/leonid/work/jb/ktor/ktor-http/ktor-http-cio/common/src/io/ktor/http/cio/internals/AsciiCharTree.kt:9
#8  kfun:io.ktor.http.cio.internals.AsciiCharTree.Node#<init>(kotlin.Char;kotlin.collections.List<1:0>;kotlin.collections.List<io.ktor.http.cio.internals.AsciiCharTree.Node<1:0>>){} (_this=0xd34a60, ch=46, exact=0xd34960, 
```

Tests are freezes in 
```
#1  0x0000000000a51972 in kotlin::mm::SuspendThreads() ()
```

**Pull request closed without merging**